### DISCUSSION TITLE
forwarding port

### DISCUSSION THREAD

**Discussion by tcurdt** [grblHAL/core#315] [Q&A]
I am trying to forward to the network connection (using `socat`) but I am having trouble to connect.

Shouldn't forwarding port tcp/23 be enough to connect (via ioSender)?

Or is there something I am missing?

**Comment by terjeio** [comment]
I do not know anything about socat...
What are you trying to do?
Forward port 23 on a grblHAL controller to port 23 on another machine in order to get through a firewall?
Or from serial to telnet or vice versa?
More details please.

**Comment by tcurdt** [comment]
I have ioSender running in a VM (via UTM). The VM can talk to the host but not the IP of grblHAL.
That's why I tried to just forward port tcp/23 to make it accessible from the host.

Now, there are a  couple of options.
I could switch the networking of the VM.
Adjust the routing or firewall.
But this goes way beyond a grblHAL question.

A similar scenario:
Some software can only deal with serial ports and does not know how to talk via tcp socket.
This could again be quite straight forward - if tcp/23 is the only thing that is needed I could proxy this to a virtual port.
But here I again had some problems.

So before I dig deeper (and provide more details)...
The only question that would be great to have answered at this stage:

Should access to port tcp/23 be enough?

**Reply by dresco** [reply]
> Should access to port tcp/23 be enough?

I would say yes, I've successfully used socat for testing network access here. On a Linux box, using CNCjs software that can only talk over serial port.

Just checked my notes, and I was using these options `socat PTY,link=./virtual-tty,raw,waitslave tcp:192.168.100.200:23`
Haven't tried it with ioSender though, as have network connectivity from the VM when running that..
